notebook-images/base/jupyter-datascience-notebook-imagestream.yaml

```diff
@@ -13,6 +13,19 @@ spec:
   lookupPolicy:
     local: true
   tags:
+  # N Version of the image
+  - annotations:
+      opendatahub.io/notebook-software: '[{"name":"Python","version":"v3.9"}]'
+      opendatahub.io/notebook-python-dependencies: '[{"name":"Boto3","version":"1.26"},{"name":"Kafka-Python","version":"2.0"},{"name":"Matplotlib","version":"3.6"},{"name":"Numpy","version":"1.24"},{"name":"Pandas","version":"1.5"},{"name":"Scikit-learn","version":"1.2"},{"name":"Scipy","version":"1.10"}]'
+      openshift.io/imported-from: quay.io/opendatahub/workbench-images
+      opendatahub.io/workbench-image-recommended: 'true'
+    from:
+      kind: DockerImage
+      name: quay.io/opendatahub/workbench-images:jupyter-datascience-ubi9-python-3.9-2023a-weekly
+    name: "py3.9-v2"
```

Comment:
use the image-tag as the imagestream tag
`name: jupyter-datascience-ubi9-python-3.9-2023a-weekly`

In my opinion, such weekly tags are bad practice in deployment references. But, you could implement a process to get the sha256 digest value each week and then update tag.from.name. I do see why tags get updated to link to new images in CICD, just do not use them in tag.from.name

Reply:
See https://github.com/opendatahub-io/odh-manifests/pull/746#issuecomment-1484430943 about using digest image hashes odh-manifests releases

Reply:
Totally fine and understood. The imagestream (not image reference) tag itself, though, should be informative as to the image-tag referenced. It all depends on what you want to achieve with upstream RHOSDS check of env variable JUPYTER_IMAGE.